example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/GooglePayFragment.kt

```diff
@@ -0,0 +1,190 @@
+
+        loadGooglePayButton()
+    }
+
+    private fun onViewState(state: GooglePayViewState) {
+        when (state) {
+            is GooglePayViewState.Error -> {
+                binding.errorView.isVisible = true
+                binding.errorView.text = getString(state.message)
+                binding.componentView.isVisible = false
+                binding.progressIndicator.isVisible = false
+                binding.googlePayButton.isVisible = false
+            }
+
+            GooglePayViewState.Loading -> {
+                binding.errorView.isVisible = false
+                binding.componentView.isVisible = false
+                binding.progressIndicator.isVisible = true
+                binding.googlePayButton.isVisible = false
+            }
+
+            GooglePayViewState.ShowButton -> {
+                binding.errorView.isVisible = false
+                binding.componentView.isVisible = false
+                binding.progressIndicator.isVisible = false
+                binding.googlePayButton.isVisible = true
+            }
+
+            GooglePayViewState.ShowComponent -> {
+                binding.errorView.isVisible = false
+                binding.componentView.isVisible = true
+                binding.progressIndicator.isVisible = false
+                binding.googlePayButton.isVisible = false
+            }
+        }
+    }
+
+    private fun onEvent(event: GooglePayEvent) {
+        when (event) {
+            is GooglePayEvent.CheckAvailability -> checkAvailability(event.paymentMethod, event.callback)
+            is GooglePayEvent.AdditionalAction -> googlePayComponent?.handleAction(event.action, requireActivity())
+            is GooglePayEvent.PaymentResult -> onPaymentResult(event.result)
+        }
+    }
+
+    private fun checkAvailability(paymentMethod: PaymentMethod, callback: ComponentAvailableCallback) {
+        GooglePayComponent.PROVIDER.isAvailable(
+            applicationContext = requireActivity().application,
+            paymentMethod = paymentMethod,
+            configuration = checkoutConfigurationProvider.getGooglePayConfiguration(),
+            callback = callback,
+        )
+    }
+
+    private fun onPaymentResult(result: String) {
+        Toast.makeText(requireContext(), result, Toast.LENGTH_SHORT).show()
+        dismiss()
+    }
+
+    private fun loadGooglePayButton() {
+        val allowedPaymentMethods = googlePayComponent?.getGooglePayButtonParameters()?.allowedPaymentMethods.orEmpty()
+        val buttonOptions = ButtonOptions
+            .newBuilder()
+            .setButtonType(ButtonType.PAY)
+            .setAllowedPaymentMethods(allowedPaymentMethods)
+            .build()
+        binding.googlePayButton.initialize(buttonOptions)
+
+        binding.googlePayButton.setOnClickListener {
+            googlePayComponent?.startGooglePayScreen(requireActivity(), ACTIVITY_RESULT_CODE)
+        }
+    }
+
+    @Deprecated("Deprecated in Java")
```

Comment:
Should we put a comment to explain why we have to use `onActivityResult`?

Reply:
Done